protocol/codec_test.go

```diff
@@ -199,3 +199,24 @@ func TestEncodeJSON(t *testing.T) {
 	require.True(t, reflect.DeepEqual(v, nsv))
 	require.True(t, reflect.DeepEqual(v, sv))
 }
+
+func TestRandomizeObjectWithPtrField(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	type testObjA struct {
+		U64 uint64
+	}
+	type testObjB struct {
+		U16  uint16
+		ObjA *testObjA
+	}
+
+	obj, err := RandomizeObject(&testObjB{})
+	require.NoError(t, err)
+	objB, ok := obj.(*testObjB)
+	require.True(t, ok)
+	t.Logf("%+v %+v", objB, objB.ObjA)
+	require.NotZero(t, objB.U16)
```

Comment:
I know it is a low probability, but this might fail from time to time.
can we make it more deterministic?

Reply:
because 0 might be chosen as a random integer?

Reply:
yes.
and on u16 it might even happen

Reply:
OK! I could also just change the uint16 to a uint64 and achieve the same result 😆

Reply:
I thought you wanted to test another type.

Reply:
OK, updated!